### DISCUSSION TITLE
Seeking your input on TGZ processing

### DISCUSSION THREAD

**Discussion by simulot** [simulot/immich-go#109] [General]
The request for reading a set of TGZ files as easily than a set of ZIP comes to me frequently. (#96, #36)

The ease of handling ZIP files without decompressing  them is currently missing for TGZ files. This feature relies on the ability to have a complete view of the archive content before  uploading accurate data into Immich. This poses a challenge for TGZ files.

I have worked on a proof of concept in the branch `FRQ--Support-tgz-takeout-files-#96-#36`. The TGZ reading without prior decompression is possible at the cost of a double read of the archive. 

I want to have your input of whether to continue the TGZ handling, knowing that
- the benefit remains small for non google photos import
- there are easy way to obtain zip files, or to decompress them upfront.

What do you think about? Worth the work or should I work on other features?

Thanks for sharing your thoughts.
Simulot

**Comment by johnny2678** [comment]
I'm one of the ones that didn't read the manual and downloaded the TGZ files instead of the ZIPs. Rather than redownload, I just extracted the TGZ files and pointed immich-go to the uncompressed Takeout folder.  

Working directly with TGZ would be ok I guess, but personally, it wouldn't be high on my priority list.

Thanks for polling your users.

**Comment by linuxturtle** [comment]
Bit of a necro-reply, but I'm with @johnny2678.  I *always* download tgz files because my archive is nearly 800G with all the duplicated album photos/videos, and I don't want to deal with downloading so many files manually, since .zip files have such a small max size.  Honestly, I wouldn't use the direct tgz import even if you wrote it, because I want to extract the .tgz and maybe edit it (like remove albums I don't want to import) before I do any importing.

**Reply by simulot** [reply]
You can select 50GB chunk size for ZIP takeout type.

The current design of the import requires to read all JSON files before proceeding to the upload. This isn't possible with a TGZ archive without parsing the TGZ files twice, or decompress them on a disk.

I have in mind a "one pass" process that could be applied to TGZ files. But it requiers few immich API changes to work as expected.

**Reply by eikowagenknecht** [reply]
> You can select 50GB chunk size for ZIP takeout type.

For anyone else coming here after thinking this is not an option: The Takeout UI very stupidly displays the dropdown behind the modal form so it's capped at 4GB:

<img width="681" height="377" alt="image" src="https://github.com/user-attachments/assets/4fff98c5-4144-4856-b609-01331f31ee5e" />

Just use the keyboard to scroll down then you can use 50GB. Or select 4GB first, then open the dropdown again and you see 50GB because it scrolled down....

<img width="650" height="359" alt="image" src="https://github.com/user-attachments/assets/e1d06208-4f35-4b0e-907a-351343d75a1b" />
